src/analysis/retail/mage/shared/defensives/IceBlock.tsx

```diff
@@ -0,0 +1,57 @@
+import TALENTS from 'common/TALENTS/mage';
+import { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import {
+  absoluteMitigation,
+  buff,
+  MajorDefensiveBuff,
+} from 'interface/guide/components/MajorDefensives/MajorDefensiveAnalyzer';
+import { SpellLink } from 'interface';
+import MajorDefensiveStatistic from 'interface/MajorDefensiveStatistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+const BASE_MITIGATION = 1.0;
+
+class IceBlock extends MajorDefensiveBuff {
+  mitigation = BASE_MITIGATION;
+
+  static dependencies = {
+    ...MajorDefensiveBuff.dependencies,
+    abilityTracker: AbilityTracker,
+  };
+  protected abilityTracker!: AbilityTracker;
+
+  constructor(options: Options) {
+    super(TALENTS.ICE_BLOCK_TALENT, buff(TALENTS.ICE_BLOCK_TALENT), options);
+    this.active =
+      this.selectedCombatant.hasTalent(TALENTS.ICE_BLOCK_TALENT) &&
+      !this.selectedCombatant.hasTalent(TALENTS.ICE_COLD_TALENT);
+
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this.onDamageTaken);
+  }
+
+  onDamageTaken(event: DamageEvent) {
+    if (!this.defensiveActive) {
+      return;
+    }
+    this.recordMitigation({
+      event,
+      mitigatedAmount: absoluteMitigation(event, this.mitigation),
+    });
+  }
```

Comment:
this will not work correctly with immunities since there is no damage value for immuned damage. with ice cold being as strong as it is, people typically only take ice block to cheese things so it is probably not worth trying to evaluate what was immuned (or doing very basic checks like seeing if you immuned anything)

Reply:
for ref, this is the implementation of `absoluteMitigation`: https://github.com/WoWAnalyzer/WoWAnalyzer/blob/f9ca50914c2d122548a14b28eaf9c69478d95c2a/src/interface/guide/components/MajorDefensives/MajorDefensiveAnalyzer.tsx#L108-L112

if you pass 1.0 in as the `mitPct`, it will calculate:

```ts
const actualAmount = undefined; // probably undefined for an immuned damage event
const priorAmount = actualAmount * (1/ (1 - 1)); // = NaN? * 1 / 0 = NaN? * Infinity
return Infinity - NaN?;
```